### ISSUE TITLE
Color gradient palette for NoRespawn column

### ISSUE THREAD

**Issue by FreezeHandz** [Vanawy/tm-run-history#33]
Hi. The new update is very nice. Being able to see the NoRespawn time is awesome. It would be nice if the color of the NoRespawn time was different based on the delta threshold colors.
This is how it looks with and without the target delta column

![image1-5](https://github.com/Vanawy/tm-run-history/assets/93701993/e9ea6ec1-7ed6-454f-ae85-0aa7b23a34c8)

With only the NoRespawn column it's just a green wall where I think it's difficult to see what time is faster than another one.

I made two palettes based on the AT color. The gradients go from dark to bright.

![gradients](https://github.com/Vanawy/tm-run-history/assets/93701993/3e9cca86-7b46-4b96-8d91-453436f52237)

**Comment by FreezeHandz** [comment]
I thought that the respawn color was only green (AT color). But by playing another map I found out that it's based on the medal's colors. 
So maybe the gradients won't work

**Comment by Vanawy** [comment]
Yeah, they're based on medal colours but some variation's definitely would be nice. No respawn times definitely need some work over all. I'll keep this suggestion in mind in next update (not sure when tho 😅)

**Comment by FreezeHandz** [comment]
In my opinion, the gradient should be based on one color for example blue. To highlight if the no respawn time is above a medal, it could be displayed with a colored circle placed to the right of the no respawn time

**Comment by Vanawy** [comment]
Yeah, for Copium medals I can actually add another column (like for normal runs) and make No respawn time use gradients you suggested. Blue is not great cause PB imrovment column already uses it.

It's all a bit difficult. I plan to add more columns based on "copium time" like delta and improvment - if you enable all of them that would be a mess of course. But even with couple columns enabled there shouldn't be wall of text (numbers in our case) in one color. So if you have more ideas about that I'm all ears!

**Comment by FreezeHandz** [comment]
What's "Copium medal"? I never heard it before

**Comment by Vanawy** [comment]
> What's "Copium medal"? I never heard it before

No Respawn Timer - Copium Timer
So here we are 😅

**Comment by FreezeHandz** [comment]
Initially, I thought that using the delta threshold for the noRespawn time could work fine. But after I watched again the screenshot example with all the times, it won't work. Because even if the delta is faster, the noRespawn time could be slower than another time with a slower delta. For example times number 8 and 9.

So the gradient must be based on the noRespawn times. I don't know how you could do this but this is the idea I have so far.
Take the faster and slower time. Get their difference. Then divide that by the number of colors, let's say 8 shades. With every 1/x make a range of the division and assign the color gradient to it. So if a "Copium time" is in one range, it assigns that color to the time.

For example

- fastest noRespawn time is 40.0
- slowest noRespawn time is 44.0

The difference is 4 and the division by 8 is 0.5
Color 1 is from 40.0 to 40.5
Color 2 is from 40.501 to 41.0
...
Color 8 is from 43.501 to 44.0

And every time there's a new fastest or slowest copium time it updates the ranges
